### ISSUE TITLE
The python version requirement has become a major problem for many devices that don't work.

### ISSUE THREAD

**Issue by aoom** [exo-explore/exo#206]
The default python version for many devices does not meet exo's expectations. Updating python causes a host of other problems. If the dependency on python version could be reduced, exo would work perfectly on more devices!

**Comment by AlexCheema** [comment]
The requirement is there because exo uses asyncio, which has bugs on older versions of Python.

Which devices can't run Python>=3.12?

On another note, I've been working on a rust node implementation which should fix these problems.

**Comment by jcobol** [comment]
> The default python version for many devices does not meet exo's expectations. Updating python causes a host of other problems. If the dependency on python version could be reduced, exo would work perfectly on more devices!

Have you tried running conda? With it, you can create as many Python environments as you like with whichever Python version(s) you need. Once created, you activate the environment and subsequent operations use that Python (including pip install). This lets you run different Python versions without interfering with the default OS version of Python.

**Comment by nmstoker** [comment]
I'm not convinced this issue is a realistic concern and the limited points raised suggest to me inexperience / poor approaches

@aoom would you care to make your case?

**Comment by TREE-Ind** [comment]
+1 for providing a basic conda env file / instructions.

The main case from our testing is that we have a number of Linux machines with various flavors of Ubuntu some of which come installed Python versions below 3.12.  Trying to mess with the baked in version will usually result in a bad day.

We've successfully used conda to get exo going on these machines, but having one for new users may help ease setup on a lot of Linux devices.

**Comment by nmstoker** [comment]
Yes, never mess with the system python version in order to run a python user script/app!

**Comment by FlorianHeigl** [comment]
+1 too, there's like 0.01% of humanity who can survive in the python ecosystem on their own wits.
